src/torchcodec/decoders/_decoder_utils.py

```diff
@@ -18,18 +19,32 @@
 
 
 def create_decoder(
-    *, source: Union[str, Path, bytes, Tensor], seek_mode: str
+    *,
+    source: Union[str, Path, io.RawIOBase, io.BufferedReader, bytes, Tensor],
+    seek_mode: str,
 ) -> Tensor:
     if isinstance(source, str):
         return core.create_from_file(source, seek_mode)
     elif isinstance(source, Path):
         return core.create_from_file(str(source), seek_mode)
+    elif isinstance(source, io.RawIOBase) or isinstance(source, io.BufferedReader):
+        return core.create_from_file_like(source, seek_mode)
     elif isinstance(source, bytes):
         return core.create_from_bytes(source, seek_mode)
     elif isinstance(source, Tensor):
         return core.create_from_tensor(source, seek_mode)
+    elif isinstance(source, io.TextIOBase):
+        raise TypeError(
+            "source is of type io.TextIOBase; did you forget to specify binary reading?"
+        )
+    elif hasattr(source, "read") and hasattr(source, "seek"):
+        # This check must be after checking for text-based reading. Also placing
+        # it last in general to be defensive: hasattr is a blunt instrument. We
+        # could use the inspect module to check for methods with the right
+        # signature.
+        return core.create_from_file_like(source, seek_mode)
 
     raise TypeError(
         f"Unknown source type: {type(source)}. "
-        "Supported types are str, Path, bytes and Tensor."
+        "Supported types are str, Path, io.RawIOBase, io.BufferedReader, bytes and Tensor."
```

Comment:
I think we should say "supported types are ... or a file-like object with read(...) and seek(...) methods"
(fill-in the ... parts).

Because of the way our input check is set up above, I don't think users will hit e.g. the "you're missing a `read` method" error - which is OK as long as this error is sufficiently informative